strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -204,6 +225,12 @@ def init_circuit(self, prog):
             if np.any(isitnew):
                 # Operation parameters
                 pars = cmd.op.p
+                # Check if any of the preparations rely on symbolic quantities
+                if isinstance(cmd.op, non_gauss_preps) and parameter_checker(pars):
```

Comment:
I didn't actually know that you could check `isinstance` with several types at the same time. 😄 🚀